src/server/game/AI/ScriptedAI/ScriptedCreature.h

```diff
@@ -210,11 +210,16 @@ struct TC_GAME_API ScriptedAI : public CreatureAI
     //Plays a sound to all nearby players
     void DoPlaySoundToSet(WorldObject* source, uint32 soundId);
 
-    //Drops all threat to 0%. Does not remove players from the threat list
-    void DoResetThreat();
-
-    float DoGetThreat(Unit* unit);
-    void DoModifyThreatPercent(Unit* unit, int32 pct);
+    // Add specified amount of threat directly to victim (ignores redirection effects) - also puts victim in combat and engages them if necessary
+    void AddThreat(Unit* victim, float amount, Unit* who = nullptr) { (who ? who : me)->AddThreat(victim, amount, nullptr, true, true); }
```

Comment:
I'm not sure what the benefit would be. Are we intending to remove the `Creature.h` include?

If not for that, this is the perfect candidate for inlining (so we save call overhead since compiler is just gonna inline it).